src/utils/queueUtils.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * src/utils/queueUtils.js
+ */
+const debug = require('debug')('refocus-collector:queueUtils');
+const logger = require('winston');
+const Queue = require('buffered-queue');
+const errors = require('../errors');
+let queueObject;
+const queueListObject = {};
+
+/**
+ * Create Queue using Buffered Queue Package
+ *
+ * @param  {String} name          Name of Queue
+ * @param  {Integer} size          Size of Queue
+ * @param  {Integer} flshTimeout   Timeout time in millisecond
+ * @param  {Boolean} verbose       Verbose flag for logging
+ * @param  {Function} flushFunction Flush function when Queue flushes
+ * @param  {Object}  refocusInstanceObj Refocus Instance Object to send Data
+ */
+function createQueue(name, size, flushTimeout, verbose,
+  flushFunction, refocusInstanceObj=null) {
```

Comment:
Looks like this function is taking in 6 parameters and it will be difficult for the caller of this function to remember the order in which the params need to be passed to this function. What about passing in an object with these properties instead?

Reply:
Anyway you need to check parameters so I am fine with any flow. I dont have any inclination.